core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go

```diff
@@ -774,9 +800,526 @@ func setupSimulatedBackendAndAuth(t *testing.T) (*simulated.Backend, *bind.Trans
 	return simulatedBackend, auth
 }
 
+// Benchmark Results:
+// Benchmark_CCIPReader_CommitReportsGTETimestamp/FirstLogs_0_MatchLogs_0-14              1000000000          0.001046 ns/op        0 B/op          0 allocs/op
+// Benchmark_CCIPReader_CommitReportsGTETimestamp/FirstLogs_1_MatchLogs_10-14            1000000000          0.001074 ns/op        0 B/op          0 allocs/op
+// Benchmark_CCIPReader_CommitReportsGTETimestamp/FirstLogs_10_MatchLogs_100-14          1000000000          0.006228 ns/op        0 B/op          0 allocs/op
+// Benchmark_CCIPReader_CommitReportsGTETimestamp/FirstLogs_100_MatchLogs_10000-14       1000000000          0.5138 ns/op          0 B/op          0 allocs/op
+func Benchmark_CCIPReader_CommitReportsGTETimestamp(b *testing.B) {
+	tests := []struct {
+		logsInsertedFirst        int
+		logsInsertedToMatchQuery int
+	}{
+		{0, 0},
+		{1, 10},
+		{10, 100},
+		{100, 10_000},
+	}
+
+	for _, tt := range tests {
+		b.Run(fmt.Sprintf("FirstLogs_%d_MatchLogs_%d", tt.logsInsertedFirst, tt.logsInsertedToMatchQuery), func(b *testing.B) {
+			benchmarkCommitReports(b, tt.logsInsertedFirst, tt.logsInsertedToMatchQuery)
+		})
+	}
+}
+
+func benchmarkCommitReports(b *testing.B, logsInsertedFirst int, logsInsertedToMatchQuery int) {
+	ctx := testutils.Context(b)
+	sb, auth := setupSimulatedBackendAndAuth(b)
+
+	// Initialize test setup
+	ts := testSetup(
+		ctx,
+		b,
+		chainS1,
+		chainS1,
+		nil,
+		evmconfig.DestReaderConfig,
+		nil,
+		nil,
+		false,
+		sb,
+		auth,
+		true)
+
+	err := ts.extendedCR.Bind(ctx, []types.BoundContract{
+		{
+			Address: ts.contractAddr.String(),
+			Name:    consts.ContractNameOffRamp,
+		},
+	})
+	require.NoError(b, err)
+
+	// Insert logs in two phases based on parameters
+	if logsInsertedFirst > 0 {
+		populateDatabaseForCommitReportAccepted(b, ts, chainS1, logsInsertedFirst, 0)
+	}
+
+	queryTimestamp := time.Now()
+
+	if logsInsertedToMatchQuery > 0 {
+		populateDatabaseForCommitReportAccepted(b, ts, chainS1, logsInsertedToMatchQuery, logsInsertedFirst)
+	}
+
+	// Reset timer to measure only the query time
+	b.ResetTimer()
+
+	// Execute the query
+	_, err = ts.extendedCR.ExtendedQueryKey(
```

Comment:
Hmm, why don't we call CCIPReader instead? Now if someone changes how the query is built by ChainReader it's not gonna be reflected in tests, right?

Reply:
CCIP reader is using some other logic. I wanted to isolate as much as possible the query. The downside is when the query is changed. It needs to be changed also in this test.